MIPS/mip-zkapps.md

```diff
@@ -0,0 +1,960 @@
+---
+mip: 0002
+title: Add zkApps to the protocol
+description: This MIP adds programmable smart contracts (zkApps) to the Mina protocol
+authors: Deepthi Kumar (deepthiskumar), Paul Steckler (@psteckler), Brett Carter (@carterbrett), Brandon Kase (@bkase)
+discussions-to:
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-02-14
+---
+
+## Abstract
+
+This MIP proposes adding programmable smart contracts, called zkApps, to the Mina protocol.
+
+## Motivation
+
+The zkApps protocol aims to add a smart contract layer to Mina which extends the
+potential of zero-knowledge cryptography by enabling the following
+characteristics while preserving the succinctness of Mina:
+
+
+  **General programmability** - able to execute and settle arbitrary programs,
+  not constrained to a particular VM model, but instead flexibly designed to
+  support any number of execution models, VM-like or otherwise
+
+  **Programmable privacy** - the privacy of both inputs to smart contracts and
+  their state can be programmed by their developers
+
+  **Constant in verification time** - individual transactions are executed, or
+  more accurately “proven” asynchronously off-chain and verified on chain in
+  time proportional to the account updates, independent of computational
+  complexity of any individual proof inside of each account update
+
+
+This MIP represents the first iteration of such a protocol.
+
+## Specification
+
+The following description summarizes the planned changes to the protocol. Not
+all technical details are described here.
+
+### Ledger account changes
+
+Recall that accounts are identified by a public key and token id pair.
+
+To accommodate zkApps, Mina ledger accounts have the following new field:
+
+- `token_symbol`: a string, up to six characters long, that can provide a name
+     for a `token_id` whose owner is the public key and token id in this
+     account. In the existing mainnet, only the default MINA `token_id` is
+     currently used. New `token_id`s are derived from the the addresses of the
+     smart contract that defines them.
+
+The following account field is removed:
+
+- `token_permissions`: this field was meant for non-default tokens, which were
```

Comment:
should the phrasing be changed here? we are introducing new tokens right? Just in a different way than before I guess?